pkg/kubelet/dockershim/checkpoint_store_test.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockershim
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+const (
+	testPath = "/tmp/testFileStore"
+)
+
+func TestFileStore(t *testing.T) {
+	cleanUpTestPath(t)
+	defer cleanUpTestPath(t)
+	testCheckpointStore(t, &FileStore{path: testPath})
+}
+
+func TestMemStore(t *testing.T) {
+	testCheckpointStore(t, NewMemStore())
+}
+
+func testCheckpointStore(t *testing.T, store CheckpointStore) {
+	var err error
+	Checkpoints := []struct {
+		key  string
+		data string
+	}{
+		{
+			"id1",
+			"data1",
+		},
+		{
+			"id2",
+			"data2",
+		},
+	}
+
+	// Test Add Checkpoint
```

Comment:
This tests that everything works.  It doesn't test that things don't not-work.  Are there error or edge cases you can test more aggressively?